pkg/crossplane/service_mariadb_database.go

```diff
@@ -96,7 +96,16 @@ func (msb MariadbDatabaseServiceBinder) Bind(ctx context.Context, bindingID stri
 		return nil, err
 	}
 
-	creds := createCredentials(endpoint, bindingID, pw, msb.instance.ID())
+	mp := 0
+	mbyte, ok := secret.Data[MetricsPortKey]
+	if ok {
+		mp, err = strconv.Atoi(string(mbyte))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse metricsPort: %w", err)
+		}
+	}
```

Comment:
Is there a need to convert the port to int? IMO it can just be keept as string.

Reply:
Fair enough. 

One could ague that failing quickly if it is not an int is a feature. But a log message in the broker is probably not helpful either..